cilium/datadog_checks/cilium/cilium.py

```diff
@@ -0,0 +1,124 @@
+    'cilium_policy_l7_forwarded_total': 'policy.l7_forwarded.total',
+    'cilium_policy_l7_parse_errors_total': 'policy.l7_parse_errors.total',
+    'cilium_policy_l7_received_total': 'policy.l7_received.total',
+    'cilium_policy_max_revision': 'policy.max_revision',
+    'cilium_policy_regeneration_time_stats_seconds': 'policy.regeneration_time_stats.seconds',
+    'cilium_policy_regeneration_total': 'policy.regeneration.total',
+    'cilium_process_cpu_seconds_total': 'process.cpu.seconds.total',
+    'cilium_process_max_fds': 'process.max_fds',
+    'cilium_process_open_fds': 'process.open_fds',
+    'cilium_process_resident_memory_bytes': 'process.resident_memory.bytes',
+    'cilium_process_start_time_seconds': 'process.start_time.seconds',
+    'cilium_process_virtual_memory_bytes': 'process.virtual_memory.bytes',
+    'cilium_process_virtual_memory_max_bytes': 'process.virtual_memory.max.bytes',
+    'cilium_subprocess_start_total': 'subprocess.start.total',
+    'cilium_triggers_policy_update_call_duration_seconds': 'triggers_policy.update_call_duration.seconds',
+    'cilium_triggers_policy_update_folds': 'triggers_policy.update_folds',
+    'cilium_triggers_policy_update_total': 'triggers_policy.update.total',
+    'cilium_unreachable_health_endpoints': 'unreachable.health_endpoints',
+    'cilium_unreachable_nodes': 'unreachable.nodes',
+    'cilium_event_ts': 'event_timestamp',
+}
+
+OPERATOR_METRICS = {
+    'cilium_operator_process_cpu_seconds_total': 'operator.process.cpu.seconds',
+    'cilium_operator_process_max_fds': 'operator.process.max_fds',
+    'cilium_operator_process_open_fds': 'operator.process.open_fds',
+    'cilium_operator_process_resident_memory_bytes': 'operator.process.resident_memory.bytes',
+    'cilium_operator_process_start_time_seconds': 'operator.process.start_time.seconds',
+    'cilium_operator_process_virtual_memory_bytes': 'operator.process.virtual_memory.bytes',
+    'cilium_operator_process_virtual_memory_max_bytes': 'operator.process.virtual_memory_max.bytes',
+    # TODO: ENI metrics are not listed
+}
+
+
+class CiliumCheck(OpenMetricsBaseCheck):
+    """
+    Collect Cilium metrics from Prometheus endpoint
+    """
+
+    def __init__(self, name, init_config, instances):
+        super(CiliumCheck, self).__init__(name, init_config, instances=self._generate_instances(instances))
+
+    def _generate_instances(self, instances):
+        """
+        Set up Cilium instance so it can be used in OpenMetricsBaseCheck
+        """
+        generated_instances = []
+
+        for instance in instances:
+            endpoint = None
+            metrics = None
+            agent_endpoint = instance.get('agent_endpoint')
+            operator_endpoint = instance.get('operator_endpoint')
+
+            # Cannot have both cilium-agent and cilium-operator metrics enabled
+            if agent_endpoint and operator_endpoint:
+                ConfigurationError("Only one endpoint needs to be specified")
```

Comment:
```suggestion
                ConfigurationError("Only one endpoint can be specified in an instance")
```

Question: Would it not make sense to have both endpoint in the same instance (like istio for example) ?

Reply:
Especially knowing that `agent_endpoint` is documented as `required` in the config

Reply:
I'm not sure if it'd make sense to have both endpoints in the same instance because one endpoint is per Cilium pod vs the other is cluster-wide.